src/sentry/sentry_metrics/indexer/ratelimiters.py

```diff
@@ -0,0 +1,136 @@
+from typing import Mapping, Optional, Sequence, Tuple
+
+from django.conf import settings
+
+from sentry import options
+from sentry.ratelimits.sliding_windows import (
+    GrantedQuota,
+    Quota,
+    RedisSlidingWindowRateLimiter,
+    RequestedQuota,
+    Timestamp,
+)
+from sentry.sentry_metrics.configuration import UseCaseKey
+from sentry.sentry_metrics.indexer.base import KeyCollection, KeyResult
+from sentry.utils import metrics
+
+writes_limiter = RedisSlidingWindowRateLimiter(
+    **settings.SENTRY_METRICS_INDEXER_WRITES_LIMITER_OPTIONS
+)
+
+
+OrgId = int
+
+
+def _build_quota_key(use_case_id: UseCaseKey, org_id: Optional[OrgId] = None) -> str:
+    use_case_str = {
+        UseCaseKey.PERFORMANCE: "performance",
+        UseCaseKey.RELEASE_HEALTH: "releasehealth",
+    }[use_case_id]
+
+    if org_id is not None:
+        return f"metrics-indexer-{use_case_str}-org-{org_id}"
+    else:
+        return f"metrics-indexer-{use_case_str}-global"
+
+
+@metrics.wraps("sentry_metrics.indexer.construct_quotas")  # type: ignore
+def _construct_quotas() -> Mapping[UseCaseKey, Sequence[Quota]]:
```

Comment:
Why do we ever need to return the quotas for all the use case ids ? We already know that an instance of the indexer can deal with one use case only. I would just pass the use case id as a parameter and return just the Sequence. 
That makes the return value simpler and the code easier to follow.

Reply:
That's fair, the entire return value used to be a global const until I realized it wouldn't be properly invalidated, that's why it looks like that. Fixed.